### PR TITLE
feat: refactor to use vcluster CLI

### PR DESCRIPTION
**Pull request by lizardruss**
Fixes DSP-67

### CODE REVIEW

File: plugin.yaml
Comment (johannesfrey) [inline plugin.yaml]: https://github.com/loft-sh/vcluster/pull/1848 is currently only on main. Might want to wait for the next release before merging this?!

```suggestion
    path: https://github.com/loft-sh/vcluster/releases/download/v0.20.0-beta.7/vcluster-darwin-amd64
```

Comment (lizardruss) [reply plugin.yaml]: Good point. This is for the most part automated [here](https://github.com/loft-sh/loft-enterprise/pull/2790/commits/e642389738829d7ec208e01b7f038d0993d67c0d), so I may need to move that to a separate commit.

File: README.md
Comment (johannesfrey) [inline README.md:7]: I'm not sure about this? Will this be correct if https://github.com/loft-sh/loft-enterprise/pull/2848 is merged? Or will be no new plugin release after that?

Comment (lizardruss) [reply README.md:7]: Should be correct. It is confusing since this plugin has always used the platform version, so it will still get updated when we do platform releases as part of that release process... but now it depends on the vcluster CLI for the actual commands.. so the assets need to point to the vcluster CLI version that the platform uses.

**Review by johannesfrey — APPROVED** [review]

**Pull request merged**
